rust/headless-client/src/ipc_service.rs

```diff
@@ -540,15 +565,18 @@ impl<'a> Handler<'a> {
         // Call `set_dns` before `set_tun` so that the tunnel starts up with a valid list of resolvers.
         tracing::debug!(?dns, "Calling `set_dns`...");
         connlib.set_dns(dns);
+        let tun_span = transaction.start_child("tun", "Raise tunnel");
         let tun = self
             .tun_device
             .make_tun()
             .map_err(|e| Error::TunnelDevice(e.to_string()))?;
         connlib.set_tun(Box::new(tun));
+        tun_span.finish();
```

Comment:
I think this might also be somewhat misleading because all we do as part of `set_tun` is writing it to a channel.

Reply:
Doesn't `make_tun` run some of the platform-specific code like parts of WinTun?

Reply:
Ah yeah, I didn't see that it is covering the `make_tun` part too. It would be nice if we could just get this data from `tracing` without having to depend on our `telemetry` module everywhere.